src/host/misc.h

```diff
@@ -44,6 +44,38 @@ int ConvertToOem(const UINT uiCodePage,
                  _Out_writes_(cchTarget) CHAR* const pchTarget,
                  const UINT cchTarget) noexcept;
 
+template<typename CharT, typename Traits>
+void StringToInputEvents(std::basic_string_view<CharT, Traits> str, std::deque<std::unique_ptr<IInputEvent>>& events)
+{
+    for (const auto& ch : str)
+    {
+        auto event = std::make_unique<KeyEvent>();
+        event->SetKeyDown(true);
+        event->SetRepeatCount(1);
+        event->SetCharData(ch);
+        events.push_back(std::move(event));
+    }
+}
+
+template<typename CharT, typename Traits>
+void InputEventsToString(std::deque<std::unique_ptr<IInputEvent>>& events, std::basic_string<CharT, Traits>& str)

```

Comment:
Is there any reason for us to need to know how many input events have been consumed?

Reply:
Oh you mean because it's a mutable reference? That's an oversight - It can be made `const`.

Reply:
What i mean is... won't the consumer want to know how many events were used? That's the only way they can adequately track what is remaining...

Reply:
I've removed the two functions now.